src/utils/ucx/config.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "config.h"
+
+#include <stdexcept>
+
+#include <absl/strings/str_format.h>
+
+#include "common/nixl_log.h"
+
+namespace nixl {
+namespace ucx {
+    Config::Config() : config_ (readUcpConfig(), &ucp_config_release) {}
+
+    void
+    Config::modify (std::string_view key, std::string_view value, bool force) {
+        const char *env_val;
+        if (force) {
+            env_val = nullptr;
+        } else {
+            env_val = std::getenv (absl::StrFormat ("UCX_%s", key.data()).c_str());
+        }
+
+        value = env_val ? env_val : value;
+        const auto status = ucp_config_modify (config_.get(), key.data(), value.data());
+        if (status != UCS_OK) {
+            NIXL_WARN << "Failed to modify UCX config: " << key << "=" << value << ": "
+                      << ucs_status_string (status);
+        } else {
+            NIXL_DEBUG << "Applied UCX config from " << (env_val ? "env var" : "NIXL") << ": "
+                       << key << "=" << value;
```

Comment:
why do we need to apply env var using config API? I suggest to do that only if no env val is present or force flag is provided

Reply:
I think the code already does what you suggested.

Please note that this code was added here https://github.com/ai-dynamo/nixl/pull/459.

Reply:
it was overlooked there, so i suggest to remove it here, since you are modifying this code anyway

Reply:
Fixed.